src/io/binary_parser.cpp

```diff
@@ -183,6 +176,26 @@ long long read_factors(string filename, dd::FactorGraph &fg) {
       }
       fg.variables[variable_id].tmp_factor_ids.push_back(fg.c_nfactor);
     }
+
+    switch (type) {
+      case (dd::FUNC_SPARSE_MULTINOMIAL): {
+        long n_weights = 0;
+        file.read((char *)&n_weights, 8);
+        n_weights = be64toh(n_weights);
+        for (long j = 0; j < n_weights; j++) {
+          file.read((char *)&weightid, 8);
+          weightid = be64toh(weightid);
+          fg.factors[fg.c_nfactor].weight_ids.push_back(weightid);
```

Comment:
Let's also turn `weight_ids` into an array since we know `n_weights` already.  Or we should at least use [`reserve`](http://en.cppreference.com/w/cpp/container/vector/reserve).